subsys/bluetooth/host/Kconfig

```diff
@@ -96,10 +96,10 @@ config BT_RX_STACK_SIZE
 	depends on BT_HCI_HOST || BT_RECV_IS_RX_THREAD
 	default 1024
 	default 2048 if BT_MESH
-	default 512 if CONFIG_BT_HCI_RAW
-	range 1024 65536
+	default 512 if BT_HCI_RAW
+	range 512 65536 if BT_HCI_RAW
 	range 1100 65536 if BT_MESH
-	range 512 65536 if CONFIG_BT_HCI_RAW
+	range 1024 65536
 	help
```

Comment:
I took a look at building samples/bluetooth/handsfree, which is the only sample that enables BT_BREDR=y and it results in BT_L2CAP_RX_MTU getting the value 200 (building for the default qemu target). In subsys/bluetooth/host/Kconfig the definition looks like follows:

`config BT_L2CAP_RX_MTU`
`        int "Maximum supported L2CAP MTU for incoming data"`
`        default 23`
`        default 65 if BT_SMP`
`        default 200 if BT_BREDR`
`        range 23 1300`
`        range 65 1300 if BT_SMP`
`        help`
`          Maximum size of each incoming L2CAP PDU.`

So at least here the default entries get process using a "last one takes precedence" rule, meaning perhaps range entries are evaluated inversely. Anyway, since you showed that this clearly works for you I'll approve the patch.